clang/lib/Parse/ParseDecl.cpp

```diff
@@ -2509,8 +2509,9 @@ Parser::DeclGroupPtrTy Parser::ParseDeclGroup(ParsingDeclSpec &DS,
 
       // P2718R0 - Lifetime extension in range-based for loops.
       if (getLangOpts().CPlusPlus23) {
-        auto &LastRecord = Actions.ExprEvalContexts.back();
+        auto &LastRecord = Actions.currentEvaluationContext();
         LastRecord.InLifetimeExtendingContext = true;
+        LastRecord.RebuildDefaultArgOrDefaultInit = true;
```

Comment:
Why is this in C++23 only?  I would expect this to apply to older stuff as well, particularly since this is a DR, right?

Reply:
Because it used for P2718R0 - Lifetime extension in range-based for loops. It's a C++23 feature. we also need to rebuild default-arg and default-init when we collect temporaries in a for-range-init.

Reply:
Ah, i see that now.  Thanks for the clarification!